File: mobile/lib/core/entities/order.dart
import 'address.dart';
import 'entity.dart';
import 'order_item.dart';
import 'order_status.dart';

class Order implements Entity {
  late final String id;
  late final int orderNumber;
  late final List<OrderItem> items;
  late final OrderStatus status;
  late final Address shippingAddress;
  late final double total;

  Order.lazy({
    String? id,
    int? orderNumber,
    List<OrderItem>? items,
    OrderStatus? status,
    Address? shippingAddress,
    double? total,
  }) {
    if (id != null) this.id = id;
    if (orderNumber != null) this.orderNumber = orderNumber;
    if (items != null) this.items = items;
    if (status != null) this.status = status;
    if (shippingAddress != null) this.shippingAddress = shippingAddress;
    if (total != null) this.total = total;
  }
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/entity/Order.kt
package me.lucasbatista.vienna.sdk.entity

class Order(
    id: String? = null,
    orderNumber: Int? = null,
    items: List<OrderItem>? = null,
    status: OrderStatus? = null,
    shippingAddress: Address? = null,
    total: Double? = null,
) {
    lateinit var id: String
    var orderNumber: Int = 0
    lateinit var items: List<OrderItem>
    lateinit var status: OrderStatus
    lateinit var shippingAddress: Address
    var total = 0.0

    init {
        id?.let { this.id = id }
        orderNumber?.let { this.orderNumber = orderNumber }
        items?.let { this.items = items }
        status?.let { this.status = status }
        shippingAddress?.let { this.shippingAddress = shippingAddress }
        total?.let { this.total = total }
    }
}
